l10n_ec_account_edi/models/sri_key_type.py

```diff
@@ -222,3 +224,22 @@ def new_range():
         ctx.sign(signature)
         ctx.verify(signature)
         return etree.tostring(doc, encoding="UTF-8", pretty_print=True).decode()
+
+    def days_to_expire(self):
+        if self.expire_date:
+            return (self.expire_date - fields.Date.context_today(self)).days
+        return 0
+
+    def action_email_notification(self):
+        certificates = self.env["sri.key.type"].search([("state", "=", "valid")])
```

Comment:
The function is executed on the same model, so you can use `self.search` directly.
```suggestion
        certificates = self.search([("state", "=", "valid")])
```